source/System.cpp

```diff
@@ -426,6 +426,15 @@ void System::Load(const DataNode &node, Set<Planet> &planets)
 	// Systems without an asteroid belt defined default to a radius of 1500.
 	if(belts.empty())
 		belts.emplace_back(1, 1500.);
+
+	// Update cached info about whether there are conditions
```

Comment:
Why the need for a cache here? What computational advantage does caching here give you?

Reply:
All Systems and Fleets have to be processed every time an event is applied, the player lands, the player departs, or a save is loaded. That adds up quickly unless the game knows not to recurse into the unaffected Systems and Fleets. Very few (if any) will have conditions in them.